src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/builders/ConstantsSummaryBuilder.java

```diff
@@ -45,25 +45,25 @@
 public class ConstantsSummaryBuilder extends AbstractBuilder {
 
     /**
-     * The maximum number of package directories shown in the constant
-     * value index.
+     * The maximum number of package directories shown in the headings of
+     * the constant values contents list and headings.
      */
-    public static final int MAX_CONSTANT_VALUE_INDEX_LENGTH = 2;
+    private static final int MAX_CONSTANT_VALUE_INDEX_LENGTH = 2;
 
     /**
      * The writer used to write the results.
      */
     protected ConstantsSummaryWriter writer;
 
     /**
-     * The set of TypeElements that have constant fields.
+     * The set of type elements that have constant fields.
      */
     protected final Set<TypeElement> typeElementsWithConstFields;
 
     /**
-     * The set of printed package headers.
+     * The set of package-group headings.
      */
-    protected final Set<PackageElement> printedPackageHeaders;
+    protected final Set<String> packageGroupHeadings;
```

Comment:
This line is the basis for the fix for item 2 that I referred to run the main description, about the inappropriate use of `PackageElement` to model the headings.